runtime/src/storage/tokio/unix.rs

```diff
@@ -29,22 +29,24 @@ impl crate::Blob for Blob {
         let mut buf = buf.into();
         let file = self.file.clone();
         task::spawn_blocking(move || {
-            file.read_exact_at(buf.as_mut(), offset)?;
+            file.read_exact_at(buf.as_mut(), offset)
+                .map_err(|e| Error::ReadFailed(Box::new(e)))?;
             Ok(buf)
         })
         .await
-        .map_err(|_| Error::ReadFailed)?
+        .map_err(|e: JoinError| Error::ReadFailed(Box::new(e)))?
     }
 
     async fn write_at(&self, buf: impl Into<StableBuf> + Send, offset: u64) -> Result<(), Error> {
         let buf = buf.into();
         let file = self.file.clone();
         task::spawn_blocking(move || {
-            file.write_all_at(buf.as_ref(), offset)?;
+            file.write_all_at(buf.as_ref(), offset)
+                .map_err(|e| Error::WriteFailed(Box::new(e)))?;
             Ok(())
         })
         .await
-        .map_err(|_| Error::WriteFailed)?
+        .map_err(|e: JoinError| Error::WriteFailed(Box::new(e)))?
```

Comment:
The type `JoinError` is probably not necessary ?

Reply:
yup removed.